hikari/guilds.py

```diff
@@ -495,6 +539,37 @@ async def fetch_self(self) -> Member:
     async def fetch_dm_channel(self) -> channels_.DMChannel:
         return await self.user.fetch_dm_channel()
 
+    async def fetch_roles(self) -> typing.Sequence[Role]:
+        """Fetch an up-to-date view of this member's roles from the API.
+
+        Returns
+        -------
+        typing.Sequence[hikari.guilds.Role]
+            An up-to-date view of this member's roles.
+
+        Raises
+        ------
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the member is not found.
+        hikari.errors.RateLimitTooLongError
+            Raised in the event that a rate limit occurs that is
+            longer than `max_rate_limit` when making a request.
+        hikari.errors.RateLimitedError
+            Usually, Hikari will handle and retry on hitting
+            rate-limits automatically. This includes most bucket-specific
+            rate-limits and global rate-limits. In some rare edge cases,
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+        member = await self.fetch_self()
```

Comment:
Member.roles is a property which retrieves the member's roles from the cache based on member.role_ids,this should be directly fetching all the roles based on member.guild_id and then filtering them based on member.role_ids